pallets/system/src/header_builder.rs

```diff
@@ -126,23 +122,45 @@ pub fn build_extension<M: Metrics>(
 	seed: Seed,
 	metrics: &M,
 ) -> HeaderExtension {
+	use once_cell::sync::Lazy;
+	static PMP: Lazy<kate::pmp::m1_blst::M1NoPrecomp> =
+		once_cell::sync::Lazy::new(|| kate::testnet::multiproof_params(256, 256));
 	use da_primitives::header::extension::{v1, v2};
 
-	let (xts_layout, commitment, block_dims, _data_matrix) = kate::com::par_build_commitments(
-		block_length.rows,
-		block_length.cols,
-		block_length.chunk_size(),
-		app_extrinsics,
+	let grid = kate::gridgen::EvaluationGrid::from_extrinsics(
```

Comment:
@Aphoh We need to merge your changes in `avail-core`, tag them and finally use here as dep.